addon/components/ember-tableish-header.js

```diff
@@ -1,21 +1,17 @@
-import Component from '@ember/component';
-import layout from '../templates/components/ember-tableish-header';
+import Component from '@glimmer/component';
+import { registerDestructor } from '@ember/destroyable';
 
-export default Component.extend({
-  layout,
-  classNames: ['ember-tableish-header'],
-
-  width: '1fr',
-
-  init() {
-    this._super(...arguments);
-
-    this.headers.addHeader(this);
-  },
+export default class EmberTableishHeader extends Component {
+  get width() {
+    return this.args.width || '1fr';
+  }
 
-  willDestroy() {
-    this._super(...arguments);
+  constructor() {
+    super(...arguments);
+    this.args.headers.addHeader(this);
 
-    this.headers.removeHeader(this);
+    registerDestructor(this, () => {
```

Comment:
glimmer components still have `willDestroy`, so what's the benefit of using `registerDestructor` instead?

Reply:
Ah youre right, the one lifecycle hook they left around!